llms/mlx_lm/utils.py

```diff
@@ -119,12 +146,33 @@ def sample(logits: mx.array) -> Tuple[mx.array, float]:
         prob = softmax_logits[0, token]
         return token, prob
 
+    if repetition_penalty and (
+        repetition_penalty < 0 or not isinstance(repetition_penalty, float)
+    ):
+        raise ValueError(
+            f"repetition_penalty must be a non-negative float, got {repetition_penalty}"
+        )
+
     y = prompt
     cache = None
+
+    repetition_context = prompt.tolist()
```

Comment:
@vishal-14069 I am a bit concerned if the prompt is very large. For instance, having too many tokens might lead to problems with text generation. Perhaps it is still necessary to apply a size limit to it.

Reply:
Hmm alright, yeah that makes sense. I will revert it back.